modules/eventing/pages/eventing-curl-spec.adoc

```diff
@@ -0,0 +1,235 @@
+** params - This must be a JavaScript Object of key-value pairs. Keys must be strings, and values must be either a string, number or boolean. These will be URL encoded as HTTP request parameters and appended to the request URL.
+** Return value (response_object) - The returned value from the cURL call which captures the response of the remote HTTP server to the request made. This is a JavaScript Object containing the following fields:
+*** body - A JavaScript variable representing the content of the response body. See below for details on how the response is unmarshalled into various JavaScript variable types.
+*** status - The numeric HTTP status code.
+*** headers - A JavaScript Object of key-value pairs with key representing the header name and value representing the header content. Both key and value will be strings.
+** Exceptions Thrown - When an unexpected error occurs, a JavaScript exception of type CurlError inheriting from the JavaScript Error class will be thrown.
+
+== Bindings
+
+To access an HTTP server using cURL, the handler needs to declare a URL binding and pass the alias of the binding to curl() calls. The binding specifies the remote URL to be accessed and all calls made using such a binding are limited to descendants of the URL specified in the binding. 
+
+HTTPS is used when the URL specifies the https:// prefix. Such a link uses https for encryption of contents, and if enabled, verifies the server certificate using the underlying OS support for server certificate verification. Client certificates are not currently supported.
+
+The binding may also specify the authentication mechanism and credentials to use. Basic, Digest and Bearer authentication methods are supported. It is strongly recommended that when authentication is used, the binding uses only https protocol to ensure credentials are encrypted when transmitted.
+
+Cookie support may be enabled at binding level if desired when accessing controlled and trusted endpoints.
+
+== Example
+
+In the below example, a cURL request is created to the specified binding profile_svc_binding with the sub-URL /person with URL parameters action and id and the body being a JSON object. The response is a JSON object and is seen containing a field profile_id. In this example, the request is automatically encoded as application/json and response is automatically parsed from JSON response, as no explicit encoding is specified.
+
+----
+var request = {
+    path: '/person',
+    params: {
+        'action': 'create',
+        'id': 23012
+    },
+    body: {
+        'name': 'John Smith',
+        'age': 25,
+        'state': 'CA',
+        'country': 'US',
+    }
+};
+
+var response = curl('POST', profile_svc_binding, request);
+if (response.status == 200) {
+  var profile_id = response.body.profile_id;
+  log("Successfully created profile " + profile_id);
+}
+----
+
+== Request marshalling
+
+
+[#optional-id,cols="1,1,1,1",options="header"]    
```

Comment:
Duplicate id (`optional-id`) for this table. Please either change the id to something unique, or remove the id altogether. You still need to set the `cols` and `options` attributes:

```
[cols="1,1,1,1",options="header"]
```